pkg/scheduler/scheduler_test.go

```diff
@@ -1403,6 +1403,352 @@ func TestSchedule(t *testing.T) {
+					Obj(),
+				*utiltesting.MakeWorkload("incoming", "eng-alpha").
+					Priority(0).
+					Queue("other").
+					Request(corev1.ResourceCPU, "2").
+					Obj(),
+			},
+			wantPreempted: sets.New("eng-alpha/a1", "eng-beta/b5"),
+			wantLeft: map[string][]string{
+				"other-alpha": {"eng-alpha/incoming"},
+			},
+			wantAssignments: map[string]kueue.Admission{
+				"eng-alpha/a1": {
+					ClusterQueue: "other-alpha",
+					PodSetAssignments: []kueue.PodSetAssignment{
+						{
+							Name: "main",
+							Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+								corev1.ResourceCPU: "on-demand",
+							},
+							ResourceUsage: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("1000m"),
+							},
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+				"eng-beta/b1": {
+					ClusterQueue: "other-beta",
+					PodSetAssignments: []kueue.PodSetAssignment{
+						{
+							Name: "main",
+							Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+								corev1.ResourceCPU: "on-demand",
+							},
+							ResourceUsage: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("1000m"),
+							},
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+				"eng-beta/b2": {
+					ClusterQueue: "other-beta",
+					PodSetAssignments: []kueue.PodSetAssignment{
+						{
+							Name: "main",
+							Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+								corev1.ResourceCPU: "on-demand",
+							},
+							ResourceUsage: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("1000m"),
+							},
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+				"eng-beta/b3": {
+					ClusterQueue: "other-beta",
+					PodSetAssignments: []kueue.PodSetAssignment{
+						{
+							Name: "main",
+							Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+								corev1.ResourceCPU: "on-demand",
+							},
+							ResourceUsage: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("1000m"),
+							},
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+				"eng-beta/b4": {
+					ClusterQueue: "other-beta",
+					PodSetAssignments: []kueue.PodSetAssignment{
+						{
+							Name: "main",
+							Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+								corev1.ResourceCPU: "on-demand",
+							},
+							ResourceUsage: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("1000m"),
+							},
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+				"eng-beta/b5": {
+					ClusterQueue: "other-beta",
+					PodSetAssignments: []kueue.PodSetAssignment{
+						{
+							Name: "main",
+							Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+								corev1.ResourceCPU: "on-demand",
+							},
+							ResourceUsage: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("1000m"),
+							},
+							Count: ptr.To[int32](1),
+						},
+					},
+				},
+			},
+		},
```

Comment:
Can you add a test case for `A workload is only eligible to do preemptions if it fits fully within nominal quota`?

Reply:
Done